src/main/kotlin/stringcalculator/StringNumbers.kt

```diff
@@ -0,0 +1,41 @@
+package stringcalculator
+
+data class StringNumbers(private val text: String?) {
+    var list: List<Int>
+        private set
+
+    init {
+        list = when {
+            text.isNullOrBlank() -> listOf(0)
+            Regex(REGEX_CUSTOM_DELIMITER).matches(text) -> parserInt(splitCustomDelimiter(text))
+            else -> {
+                parserInt(splitDefaultDelimiter(text))
+            }
+        }
+    }
+
+    private fun splitDefaultDelimiter(value: String): List<String> = value.split(REGEX_DEFAULT_DELIMITERS.toRegex())
+
+    private fun parserInt(list: List<String>): List<Int> {
+        return list.map { strNum ->
+            require(strNum.isInt()) { "숫자 이외의 값은 입력 받을 수 없습니다." }
+            require(strNum.toInt() >= 0) { "음수는 입력 받을 수 없습니다." }
+            strNum.toInt()
+        }
+    }
+
+    private fun splitCustomDelimiter(text: String): List<String> {
+        val result = Regex(REGEX_CUSTOM_DELIMITER).find(text) ?: throw IllegalArgumentException("검증된 패턴입니다.")
+        return result.let {
+            val customDelimiter = it.groupValues[1]
+            it.groupValues[2].split(customDelimiter)
```

Comment:
`it.groupValues[1]` ,`it.groupValues[2]` 와 같이 MatchResult에서 가지고 온다면 코드에서 빠르게 무엇을 의미하는지 체크하기가 힘들다고 생각해요.

저는 이런 경우에는 코틀린의 destructuring을  사용하고 있습니다. 사용하게 되면 아래와 같이 나올 것 같네요 🤩

```suggestion
            val (customDelimiter, numberText) = it.groupValues
            numberText.split(customDelimiter)
```

링크
- [Destructuring declarations](https://kotlinlang.org/docs/destructuring-declarations.html)

Reply:
`//;\n1;2;3`
`val (customDelimiter, numberText) = it.groupValues`
이렇게 하면 customDelimiter 값은 `//;` 로 나옵니다.

말씀해주신 destructuring 을 참고하여 아래와 같이 리팩토링해보았습니다!
`val (customDelimiter, numberText) = it.destructured`